src/main/java/org/shanerx/tradeshop/utils/configuration/SettingsController.java

```diff
@@ -0,0 +1,142 @@
+/*
+ *
+ *                         Copyright (c) 2016-2019
+ *                SparklingComet @ http://shanerx.org
+ *               KillerOfPie @ http://killerofpie.github.io
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *                http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  NOTICE: All modifications made by others to the source code belong
+ *  to the respective contributor. No contributor should be held liable for
+ *  any damages of any kind, whether be material or moral, which were
+ *  caused by their contribution(s) to the project. See the full License for more information.
+ *
+ */
+
+package org.shanerx.tradeshop.utils.configuration;
+
+import org.shanerx.tradeshop.TradeShop;
+import org.shanerx.tradeshop.utils.configuration.interfaces.YAMLConfigurationControllerInterface;
+import org.shanerx.tradeshop.utils.configuration.interfaces.YAMLConfigurationInterface;
+
+import java.util.Arrays;
+
+public class SettingsController extends YAMLConfigurationController implements YAMLConfigurationControllerInterface {
+
+    public SettingsController(TradeShop plugin) {
+        super(plugin, "config");
+        reload();
+    }
+
+    @Override
+    public YAMLConfigurationInterface findKey(String name) {
+        return SettingsEnum.valueOf(name.toUpperCase().replace("-", "_"));
+    }
```

Comment:
This should be done in the superclass, possibly by making the superclass a generic of parameterized type `<E extends Enum<?>>` and then accessing the values of the enum via something similar to  `E.valueOf(...)`, that way all these things can be generalised and used by `MessagesController` too.

Reply:
This cant be done in a superclass because it has a direct reference to the Enum, see comment on one of the unresolved enum comments for more details on that.

Reply:
We are not really extending enyms in the example. In `interface Name <E extends Enum<?>>` we are not extending an enum, we are only templating. It means E is an enum type.
In the enum declaration we would have 
`enum AnotherName implements Name<AnotherName>`